### PR TITLE
Remove "zoom_to" plugin and add ESRI search control

### PR DESCRIPTION
**Pull request by caseypt**
Replaces the `zoom_to` plugin with the ESRI search control.

image

**Testing**
- Visit the app and try out the search control in the header.
- Verify all references to the `zoom_to` plugin have been removed.

**Notes**
There are some display issues on small screens. I'm going to update https://github.com/CoastalResilienceNetwork/GeositeFramework/issues/690.

Connects to #701

### CODE REVIEW

File: src/GeositeFramework/js/Map.js [hunk 1]
```diff
@@ -7,15 +7,17 @@ require(['use!Geosite',
          'esri/map',
          'esri/layers/ArcGISTiledMapServiceLayer',
          'esri/geometry/Extent',
-         'esri/SpatialReference'
+         'esri/SpatialReference',
+         'esri/dijit/Search',
```
Comment (kdeloach) [inline src/GeositeFramework/js/Map.js]: I think we have run into issues in the past with trailing commas in some versions of IE.

**Comment by kdeloach** [comment]
This diff looks great.

**Comment by caseypt** [comment]
> I think the input control should always be open, but we can let @jfrankl weigh in.

Yeah, I wasn't sure. It's a boolean option when initializing the control, so it's easy to update.

> We'll need to get it worked into the responsive layout. Presently, it gets smashed up with the "region dropdown". Probably best to create an issue for that, and we'll work it into the design sprint when we accumulate additional tasks.

I added this to #690.

File: src/GeositeFramework/js/Map.js (hunk 1 above)
Comment (caseypt) [inline src/GeositeFramework/js/Map.js]: Thanks

**Comment by jfrankl** [comment]
To confirm what Matt said: the input should always be open.

**Comment by caseypt** [comment]
Pushed a fixup commit to ensure the search control is always open.

**Comment by caseypt** [comment]
Going to merge since this has been reviewed and approved.

**Pull request merged**